SNAKE-16 Страница форума

Totentu opened the pull request:
Разметка и базовая функциональность (на заглушках) страницы форума.

Totentu committed:
Начало разработки страницы форума.

Totentu committed:
Начало разработки страницы форума.

Totentu committed:
Начало разработки страницы форума.

Totentu committed:
Создал компонент Theme. Компонент Heading и обновление компонента App взято из другой ветки разработки (для будущей совместимости).

Totentu committed:
Улучшил разметку, добавил разметку для создания новой темы.

Totentu committed:
Создал функциональность (пока на заглушках) создания новой темы.

ndronov commented on package-lock.json:
```diff
@@ -1,8 +1,11521 @@
   "name": "snake",
   "version": "1.0.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
версия package-lock.json продолжает меняться на 2,
надо подумать, что сделать с этим

ndronov commented on src/components/App/App.tsx:
```diff
@@ -9,18 +9,24 @@ import LeaderboardPage from '../../pages/LeaderboardPage';
+          </Routes>
+        </main>
+      </BrowserRouter>
+    </div>
```
опять конфликты будут при мерже )

ndronov commented on src/components/Theme/Theme.css:
```diff
@@ -0,0 +1,65 @@
+
+.theme:hover {
+  box-shadow: 0 0 50px #CDD02A;
+
```
лишняя строка

ndronov commented on src/components/Theme/Theme.tsx:
```diff
@@ -0,0 +1,35 @@
+  content: string;
+};
+
+const Input: FC<ThemeProps> = (props) => {
```
имя осталось от Input

ndronov commented on src/components/Theme/Theme.tsx:
```diff
@@ -0,0 +1,35 @@
+  );
+};
+
+export default Input;
```
и тут

ndronov commented on src/components/Theme/Theme.tsx:
```diff
@@ -0,0 +1,35 @@
+    newElement = <div className={cn('theme-newCount')}>Новых сообщений: {newCount}</div>;
+  } else {
+    newElement = <div className={cn('theme-newCount')}> </div>;
+  }
```
можно вот так оформить:

```jsx
{newCount > 0 && <div className={cn('theme-newCount')}>{`Новых сообщений: {newCount}`}</div>
```

ndronov commented on src/pages/ForumPage/ForumPage.tsx:
```diff
@@ -1,14 +1,73 @@
+    '0' + val.getMinutes()
+  ].map((component) => component.slice(-2)); // взять последние 2 цифры из каждой компоненты
+  return tmp.slice(0, 3).join('.') + ' ' + tmp.slice(3).join(':');
+}
```
```typescript
function formatDateTime(date: Date): string {
  const DD = date.getDate().toString().padStart(2, '0');
  const MM = (date.getMonth() + 1).toString().padStart(2, '0');
  const YYYY = date.getFullYear().toString();

  const hh = date.getHours().toString().padStart(2, '0');
  const mm = date.getMinutes().toString().padStart(2, '0');

  return `${DD}.${MM}.${YYYY} ${hh}:${mm}`;
}
```

ndronov commented on src/pages/ForumPage/ForumPage.tsx:
```diff
@@ -1,14 +1,73 @@
+  const contentRef = useRef(null);
+  const [themes, setThemes] = useState(out_arr);
+
+  const CreateNewTheme = () => {
```
с маленькой буквы

ndronov commented on src/pages/ForumPage/ForumPage.tsx:
```diff
@@ -1,14 +1,73 @@
+    setThemes(
+      [
+        {
+          id: id,
```
такие конструкции можно сокращать до 
```js
id,
```

ndronov commented on src/pages/ForumPage/ForumPage.tsx:
```diff
@@ -1,14 +1,73 @@
+          author: 'Текущий пользователь',
+          mesCount: 1,
+          newCount: 0,
+          content: content
```
и тут

ndronov commented on src/pages/ForumPage/ForumPage.tsx:
```diff
@@ -1,14 +1,73 @@
+import Theme from '../../components/Theme';
 
 import './ForumPage.css';
+import cn from 'classnames';
```
импорты внешних зависимостей лучше помещать до внутренних

ndronov commented on src/components/Theme/Theme.tsx:
```diff
@@ -0,0 +1,35 @@
+import React, { FC } from 'react';
+import cn from 'classnames';
+
+import './Theme.css';
```
`Theme` - это тема форма? Если да, то лучше назвать например `Topic`.
`Theme` больше подходит для описание тёмной и светлой темы.

ndronov commented on src/pages/ForumPage/ForumPage.css:
```diff
@@ -1,3 +1,59 @@
+  overflow: auto;
+  overflow-x:hidden;
+  background: var(--transparent-blue);
+
```
лишняя строка

Totentu committed:
Подключил stylelint, поправил синтаксис стилей, заменил Theme на Topic по всему коду, еще мелкие правки.

ndronov reviewed (changes requested):
написал свой вариант `formatDateTime()`

Totentu committed:
Мелкая правка для игры (не имеет отношения к этой ветке).

Totentu committed:
Мелкая правка для игры (не имеет отношения к этой ветке).

Totentu committed:
Решены следующие задачи:
- адаптация дизайна,
- создание компонента Scroll для прокручивания того, что внутри,
- создание страницы сообщений,
- переходы из тем в сообщения и обратно,
- выделение utils.ts для общих функций.

Totentu committed:
Решены следующие задачи:
- адаптация дизайна,
- создание компонента Scroll для прокручивания того, что внутри,
- создание страницы сообщений,
- переходы из тем в сообщения и обратно,
- выделение utils.ts для общих функций.

Totentu committed:
Merge remote-tracking branch 'origin/snake-16-forum' into snake-16-forum

Totentu committed:
Восстановил package-lock.json

ndronov commented on src/components/Button/Button.css:
```diff
@@ -8,13 +8,14 @@
   padding-right: var(--gap-5);
-  color: var(--violet);
-  font-family: var(--impact-font);
+  color: rgb(0, 0, 0);
```
`color: var(--black);`

ndronov commented on src/components/Heading/Heading.css:
```diff
@@ -1,29 +1,39 @@
 .heading {
   margin: 0;
-  font-family: var(--impact-font);
+  color: rgb(255, 255, 255);
```
color: var(--white);

ndronov commented on src/components/Heading/Heading.css:
```diff
@@ -1,29 +1,39 @@
   font-size: calc(var(--gap) * 16);
-  color: var(--violet);
+  font-family: var(--title-font);
+  color: rgba(236, 239, 73, 1);
```
лучше добавить константу в colors.css

ndronov commented on src/components/Input/Input.css:
```diff
@@ -23,11 +23,11 @@
   font-family: var(--platform-font);
   font-size: 100%;
-  border-radius: var(--normal-border-radius);
+  border-radius: var(--gap5);
```
дефис пропущен: `--gap-5`

ndronov commented on src/styles/spacing.css:
```diff
@@ -12,7 +13,14 @@
+  --gap5-3: calc(var(--gap) * 3); /* 15px */
+  --gap5-4: calc(var(--gap) * 4); /* 20px */
+  --gap5-5: calc(var(--gap) * 5); /* 25px */
+  --gap5-6: calc(var(--gap) * 6); /* 30px */
```
новая система отступов?

кажется, что лучше такого избегать, можно просто поменять --gap c 4px на 5px

ndronov commented on src/components/Message/Message.tsx:
```diff
@@ -0,0 +1,25 @@
+  return (
+    <div className={cn('message', 'text-field')}>
+      <div className={cn('message-author')}>
+        {author} &nbsp;&nbsp; {datetime}
```
`&nbsp;&nbsp;` - это нормально работает?
в react часто проблемы с такими символами...

ndronov commented on src/components/Message/Message.tsx:
```diff
@@ -0,0 +1,25 @@
+};
+
+const Message: FC<MessageProps> = (props) => {
+  const { datetime, author, content } = props;
```
`dateTime`

ndronov commented on src/components/Message/Message.tsx:
```diff
@@ -0,0 +1,25 @@
+import './Message.css';
+
+type MessageProps = {
+  datetime: string;
```
`dateTime`

ndronov commented on src/components/Scroll/Scroll.css:
```diff
@@ -0,0 +1,68 @@
+.img-button:hover {
+  filter: drop-shadow(var(--box-shadow));
+}
+.scroll-upbutton {
```
`.scroll-up-button`

ndronov commented on src/components/Scroll/Scroll.css:
```diff
@@ -0,0 +1,68 @@
+  background-image: url('../../assets/up.png');
+}
+
+.scroll-downbutton {
```
`.scroll-down-button`

ndronov commented on src/components/Scroll/Scroll.tsx:
```diff
@@ -0,0 +1,115 @@
+  const upClick = () => {
+    contentRef.current.style.scrollBehavior = 'smooth';
+    contentRef.current.scrollTop -= 300;
+    contentRef.current.style.scrollBehavior = '';
```
так лучше не делать - в одной функции лучше не менять один параметр два раза
https://www.youtube.com/watch?v=cCOL7MC4Pl0

ndronov commented on src/components/Scroll/Scroll.tsx:
```diff
@@ -0,0 +1,115 @@
+  );
+};
+
+export default ScrollComponent;
```
компонент выглядит сложно, надо будет потом подумать, как его можно упростить

ndronov commented on src/components/Topic/Topic.css:
```diff
@@ -0,0 +1,47 @@
+  text-align: center;
+}
+
+.topic-mesCount {
```
лучше в стилях не мешать snake-case и camelCase

ndronov commented on src/components/Topic/Topic.tsx:
```diff
@@ -0,0 +1,50 @@
+import './Topic.css';
+
+type TopicProps = {
+  datetime: string;
```
`dateTime` - разные слова лучше отделять camelCase

ndronov commented on src/pages/ForumPage/ForumPage.tsx:
```diff
@@ -1,14 +1,65 @@
+import Topic from '../../components/Topic';
+import Scroll from '../../components/Scroll';
 
+import '../../components/TextArea/TextArea.css';
```
должно работать и без этого импорта

Totentu committed:
Внес правки в соответствии с ревью.
